### ISSUE TITLE
Gamma Hurdle DWPC model

### ISSUE THREAD

**Issue by zietzm** [greenelab/hetmech#123]
DWPCs across multiple permutations for a single source, target node combination have a zero-inflated distribution.
![image](https://user-images.githubusercontent.com/7230303/41719625-95899cc0-752e-11e8-9212-debf0d33d01d.png)

Meanwhile, the nonzero values follow roughly a gamma distribution
![image](https://user-images.githubusercontent.com/7230303/41719639-9d74f0f6-752e-11e8-8d1f-51bb84f16eb3.png)

The distributions of DWPC values are similar for a single source, target degree combination along a single metapath over all permutations. Below, orange is the distribution of nonzero values, blue is the distribution of values including zeros.

![image](https://user-images.githubusercontent.com/7230303/41727575-ae1c8c5c-7542-11e8-9d77-811d28a094ee.png)

![image](https://user-images.githubusercontent.com/7230303/41740093-014485de-7565-11e8-80cc-de59bb0795e8.png)

![image](https://user-images.githubusercontent.com/7230303/41740132-1d295900-7565-11e8-91c4-92caf8af932a.png)

[Full pdf](https://github.com/greenelab/hetmech/files/2124995/math.pdf)

In the past, we have modeled the distribution of permuted DWPC values for a single source, target combination across permutations as a simple normal distribution with the mean and variance calculated from the various permuted values. For example, in the plot above, we show the distribution of permuted DWPC values for the source, target degree combination ```(448, 6)``` across 25 permutations.

In general, the gamma hurdle model will more precisely fit the nonzero data than a cut-off normal distribution.
![image](https://user-images.githubusercontent.com/7230303/41734737-d02c4ce8-7555-11e8-8cbb-8f961a59b435.png)

**Comment by dhimmel** [comment]
This is a brilliant work by @zietzm! As I understand it, we can calculate the gamma hurdle parameters with the same summary values that we are efficiently computing currently: sum, sum of squares, and number of nonzero DWPCs.

Some random thoughts I have:

+ In the third formula, is there a missing plus sign?
+ previously we needed one nonzero value to compute the standard deviation of the normal. Now we will need two distinct nonzero DWPCs since the gamma distribution will not be fit to any of the zero values.

It would be great to fit these gamma hurdle models and see how well they approximate actual permuted DWPC distributions. We could evaluate this statistically and visually. It would be nice to evaluate a range of permuted DWPC distributions in terms of nonzero percentage and skewdness.

**Comment by zietzm** [comment]
> In the third formula, is there a missing plus sign?

No. The two probabilities are being multiplied rather than added.  

> Now we will need two distinct nonzero DWPCs 

That's correct. Hopefully that should not be a problem, now that we have 200 permutations, but potentially this could be something we have to deal with.
